uwblueprint/rowan-house
# Admin Dashboard Modals

## Implementation Description
Implements landing Page: Modals #64 
Implements modals to edit and create courses and modules, and to confirm deletes of courses/modules. Does not include backend calls to submit forms, or photo uploading for edit module modal.

Components added:
- DeleteModal.tsx: base delete modal (can be reused for different deletion purposes)
- Edit Course/Module Modals: specific modals for editing courses and modules
Common:
- TextInput.tsx: component for user input in forms (based off Chakra UI input)
- TextArea.tsx: component for user input more than one line (based off Chakra UI TextArea)
- SwitchInput.tsx: based off Chakra UI switch
- Modal.tsx: base modal component (based off Chakra UI Modal)

Still needs to be implemented:
- photo uploading for module editing
- error messages and states for when user enters invalid input (ex. empty title)

## Screenshots
![image](https://user-images.githubusercontent.com/41529410/164993895-ffe0e85e-61b5-4058-924f-2040d02b024a.png)
![image](https://user-images.githubusercontent.com/41529410/164993909-6a086d72-26d0-4103-a5c7-5682d00050d2.png)

## What should reviewers focus on?
- Are `useMutation` hooks being used properly?
- is the formatCourseRequest function alright? It's definitely an overloaded function, it might be good to use a reducer to be more explicit

## Testing Procedure
- Try invalid inputs (e.g. no inputs)
- Make sure every edit, delete, and cancel works as intended

## Things to Note
- This refetches the entire COURSES query upon a mutation which is pretty inefficient. Might be worth reconsidering. 

## Checklist
